### PR TITLE
Skip equal elements from begin and end optimization. Also trivial optimization if { from } and { to } are same objects.

### PR DESCRIPTION
**Pull request by bga**
Im noob in git history manipulation magic. Let me try again.
That optimization reduces grid size primary and also hugely reduces run time coz Levenshtein is O(n*m)
Changed only [esm/main.js] and [cjs/main.js] files but not [./index.js] nor [./min.js] coz unfortunately nmp does not work on my Win7 machine and i can not even do { npm install } to download deps and do automatic tests but i tested manually.

### CODE REVIEW

**Comment by WebReflection** [comment]
Locally, your PR branches out and for some reason coveralls is not catching that (how weird).

Am I correct in believing that after your changes on top there won't ever be a case where there are no operations to perform?

It's the only branch left out:

screenshot from 2017-11-27 17-35-47

**Comment by bga** [comment]
TravisCI log looks fine. I made manual test https://jsbin.com/hoquxiyodo/edit?html,js,console,output

**Comment by WebReflection** [comment]
I should've explained myself in a better way ... the PR looks fine, it's awesome, thank you!

**But** the code coverage leaves a branch out. The branch is that `if (length)` that is never falsy as `0`.

I believe with current changes it won't ever happen that the `length` is `0` while before, `length` equal to `0` would've meant that there were no changes to apply.

Now that use case is pre-optimize on top, right?

**Comment by WebReflection** [comment]
look, I think I'll just merge this and tweak later on on top.

Thanks

**Comment by bga** [comment]
Now i got it. I added `if(from == to) return` and your `test("same", "same")` will never reach `if(length)`. That cond can be safely removed.

**Comment by WebReflection** [comment]
so, `v2.0.0` went out with your changes in (and slightly readapted code to at least follow the rest of the convention and never go over 80 columns)
